From 949072f7c34a9f3f1169fef26ca114c452fba2c5 Mon Sep 17 00:00:00 2001
From: Jonathan Nieder <jrnieder@gmail.com>
Date: Thu, 23 Jun 2011 07:27:06 -0500
Subject: [PATCH 07/15] Makefile: do not use setgid bit on directories on
 GNU/kFreeBSD

The g+s bit on directories to make group ownership inherited is a
SysVism; BSD and most of its descendants do not need that since they
do the sane thing by default without g+s.  Because of this, on some
filesystems (but not others), the kernel of FreeBSD does not allow
non-root users to set setgid bit on directories (tmpfs works that way;
UFS does not), causing errors when one tries:

	$ git init --shared dir
	fatal: Could not make /tmp/dir/.git/refs writable by group

Since the setgid semantics are there already without the setgid bit,
it's better to avoid setting it.  Since v1.5.5-rc0~59^2 (Do not use
GUID on dir in git init --share=all on FreeBSD, 2008-03-05), git on
true FreeBSD does exactly that; so set DIR_HAS_BSD_GROUP_SEMANTICS
when $(uname -s) is GNU/kFreeBSD to make sure machines that use a GNU
userland with the kernel of FreeBSD get the same fix.

This fixes t0001-init.sh and t1301-shared-repo.sh on GNU/kFreeBSD
when --root points to a directory that uses tmpfs.

Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 Makefile |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/Makefile b/Makefile
index 8d6d451..924749e 100644
--- a/Makefile
+++ b/Makefile
@@ -820,6 +820,7 @@ ifeq ($(uname_S),GNU/kFreeBSD)
 	NO_STRLCPY = YesPlease
 	NO_MKSTEMPS = YesPlease
 	HAVE_PATHS_H = YesPlease
+	DIR_HAS_BSD_GROUP_SEMANTICS = YesPlease
 endif
 ifeq ($(uname_S),UnixWare)
 	CC = cc
-- 
1.7.5.4

